### ISSUE TITLE
`prefer-top-level-await` false positive - Calls of async functions are flagged even if used as part of a larger await statement.

### ISSUE THREAD

**Issue by RebeccaStevens** [sindresorhus/eslint-plugin-unicorn#1919]
Calls of async functions are flagged even if used as part of a larger `await` statement.

`prefer-top-level-await`

```js
await Promise.all([
  someTask(),              // This should not be flagged.
  someOtherTask()          // This should not be flagged.
]);

async function someTask() {
  return "Some Value"
}

async function someOtherTask() {
  return "Some Other Value"
}
```
